### ISSUE TITLE
Appengine OAuth2Decorator: Convert redirect address to string

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Dragon-Born/google-api-python-client#72]
```
When using the OAuth2Decorator with Python 2.7, it raises the following error:
InvalidResponseError: headers must be str

This is because self.request.get('state') in appengine.py returns a unicode 
string. This unicode string is then used as an argument for 'self.redirect', 
which causes the error. The unicode string should be converted to a string 
before it is passed on.
```

Original issue reported on code.google.com by `psch...@google.com` on 6 Dec 2011 at 3:48

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 3b633b101d6f.
```

Original comment by `jcgregorio@google.com` on 7 Dec 2011 at 3:51
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Issue 73 has been merged into this issue.
```

Original comment by `jcgregorio@google.com` on 7 Dec 2011 at 3:52
